### PR TITLE
[docs] be more specific about the definition of page endpoints

### PR DESCRIPTION
**Pull request by gtm-nayan**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by kristjanmar** [comment]
I think this is harder to understand than the current sentence and I'm not sure what it means. But I agree that the docs on page endpoints can be improved, especially with more info about how they are different from the load function.

**Comment by gtm-nayan** [comment]
@kristjanmar how about now?

**Comment by kristjanmar** [comment]
Much better.

"The filename for a page endpoint must also include any [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching) in its page's name."

Maybe something like this would be optimal just to make it more clear that this is an IF:

"If the page name has [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching), then the filename for the page endpoint also needs to include them."

File: documentation/docs/02-routing.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ The job of a [request handler](/docs/types#sveltejs-kit-requesthandler) is to re
 
 #### Page endpoints
 
-If an endpoint has the same filename as a page (except for the extension), the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR.
+If an endpoint has the same filename (except the extension) as a page, the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR. If a page has [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching) in its filename then its page endpoint's filename must also include them.
```
Comment (benmccann) [inline documentation/docs/02-routing.md:103]:
```suggestion
If an endpoint has the same filename (except the extension) as a page, the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR. If a page uses syntax for [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching) in its filename then the corresponding page endpoint's filename must also include them.
```

File: documentation/docs/02-routing.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ The job of a [request handler](/docs/types#sveltejs-kit-requesthandler) is to re
 
 #### Page endpoints
 
-If an endpoint has the same filename as a page (except for the extension), the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR.
+If an endpoint has the same filename (except the extension) as a page, the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR. If a page uses syntax for [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching) in its filename then the corresponding page endpoint's filename must also include them.
```
Comment (benmccann) [inline documentation/docs/02-routing.md:103]: I found the first sentence easier to read before
```suggestion
If an endpoint has the same filename as a page (except for the extension), the page gets its props from the endpoint — via `fetch` during client-side navigation, or via direct function call during SSR. If a page uses syntax for [named layouts](/docs/layouts#named-layouts) or [matchers](/docs/routing#advanced-routing-matching) in its filename then the corresponding page endpoint's filename must also include them.
```

**Review by Rich-Harris — APPROVED** [review]

**Comment by Rich-Harris** [comment]
much better 👍

**Pull request merged**